engine/apps/slack/models/slack_message.py

```diff
@@ -18,24 +18,48 @@
 
 if typing.TYPE_CHECKING:
     from apps.alerts.models import AlertGroup
+    from apps.base.models import UserNotificationPolicy
+    from apps.slack.models import SlackChannel
+    from apps.user_management.models import User
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
 
 class SlackMessage(models.Model):
     alert_group: typing.Optional["AlertGroup"]
+    channel: "SlackChannel"
 
     id = models.CharField(primary_key=True, default=uuid.uuid4, editable=False, max_length=36)
-
     slack_id = models.CharField(max_length=100)
 
-    # TODO: convert this to a foreign key field to SlackChannel
-    channel_id = models.CharField(max_length=100, null=True, default=None)
+    _channel_id = models.CharField(max_length=100, null=True, default=None)
+    """
+    DEPRECATED: this is no longer being referenced/set, drop in a separate PR/release
+    """
+
+    channel = models.ForeignKey(
+        "slack.SlackChannel", on_delete=models.CASCADE, null=True, default=None, related_name="slack_messages"
+    )
+    """
+    TODO: once we've migrated the data in `_channel_id` to this field, set `null=False`
+    as we should always have a `channel` associated with a message
+    """
```

Comment:
what if Slack channel gets deleted (for example [here](https://github.com/grafana/oncall/blob/14c1b37c87ada3458bbd07024dabc0480f9da3a7/engine/apps/slack/scenarios/slack_channel.py#L56-L59))? will it still be possible to use `SlackMessage.permalink` to redirect users to Slack?

`on_delete=models.CASCADE` this means if alert group was posted to `#alerts` and then `#alerts` is deleted, we will not be able to reference Slack messages from that channel in alert groups (for things like Slack permalink, resolution notes, etc.). I think we don't want that right?

Reply:
good question! I was also questioning what the right `on_delete` would be here. My thought was that if you delete a Slack channel.. everything inside the channel vanishes. So permalinks/resolution notes/etc are no longer relevant for those `SlackMessage`s (but maybe I'm overlooking something?)